lib/models/api-key.ts

```diff
@@ -253,22 +269,31 @@ const getApiKeysModel = function (
 		async getDeviceApiKeysByDevice(
 			uuidOrId: string | number,
 			options: BalenaSdk.PineOptions<BalenaSdk.ApiKey> = {},
-		): Promise<BalenaSdk.ApiKey[]> {
+		): Promise<BalenaSdk.ApiKey[] | number> {
 			const { actor } = await deviceModel().get(uuidOrId, {
 				$select: 'actor',
 			});
+			const internalOptions = {
+				$filter: {
+					is_of__actor: actor,
+				},
+			};
 
-			return await pine.get({
-				resource: 'api_key',
-				options: mergePineOptions(
-					{
-						$filter: {
-							is_of__actor: actor,
+			if (options.$count) {
+				return await pine.get({
+					resource: 'api_key',
+					options: {
+						$count: {
+							...internalOptions,
+							...options.$count,
 						},
```

Comment:
Shouldn't this be v ?
```suggestion
						$count: mergePineOptions(internalOptions, options.$count),
```